src/git/gitProviderService.ts

```diff
@@ -2440,6 +2442,22 @@ export class GitProviderService implements Disposable {
 		return promise;
 	}
 
+	@gate()
+	@log()
+	async writeAddedReposToPathMap(repos: Repository[]): Promise<void> {
+		const scope = getLogScope();
+		for (const repo of repos) {
+			try {
+				await this.container.repositoryIdentity.addFoundRepositoryToMap(
+					repo,
+					await this.container.repositoryIdentity.getRepositoryIdentity(repo),
+				);
```

Comment:
Can we hide the need to call `getRepositoryIdentity` into the `addFoundRepositoryToMap` call, or at least allow it, but have that method do it if missing.